augly/image/functional.py

```diff
@@ -1786,3 +1786,57 @@ def vflip(
     imutils.get_metadata(metadata=metadata, function_name="vflip", **func_kwargs)
 
     return imutils.ret_and_save_image(aug_image, output_path)
+
+
+def skew(
+    image: Union[str, Image.Image],
+    output_path: Optional[str] = None,
+    level: float = 1.0,
+    max_level: float = 10.0,
+    axis: str = "x_axis",
+    metadata: Optional[List[Dict[str, Any]]] = None,
+) -> Image.Image:
+    
+    """
+    Skew image with respect to X-axis or Y-axis
+    
+    @param image: the path to an image or a variable of type PIL.Image.Image
+        to be augmented
+        
+    @param output_path: the path in which the resulting image will be stored.
+        If None, the resulting PIL Image will still be returned
+
+    @param level: Level of the operation that will be in [0, The maximum value of level]
+    
+    @param max_level: Maximum value that the operation can have. This will be
+        scaled to level / PARAMETER_MAX.
+        
+    @param axis: Axis with respect to change like x-axis or y-axis
+
+    @param metadata: if set to be a list, metadata about the function execution
+        including its name, the source & dest width, height, etc. will be appended
+        to the inputted list. If set to None, no metadata will be appended or returned
+
+    @returns: the augmented PIL Image
+    """
+    
+    image = imutils.validate_and_load_image(image)
+    func_kwargs = imutils.get_func_kwargs(metadata, locals())
+    
+    level = float(np.random.uniform(low=0.1, high=level)) * max_level / 10
+    w = image.size[0]
+    h = image.size[1]
+    
+    if np.random.uniform() > 0.5:
+        level = -level
+        
+    if axis == 'x-axis':
+        aug_img = image.transform((w, h),Image.AFFINE, (1, level, 0, 0, 1, 0),
+                                resample=Image.BILINEAR)
+        imutils.get_metadata(metadata=metadata, function_name="skew", aug_img=aug_img, **func_kwargs)
+        return aug_img
+    elif axis == 'y-axis':
+        aug_img = image.transform((w, h),Image.AFFINE, (1, 0, 0, level, 1, 0),
+                                resample=Image.BILINEAR)
+        imutils.get_metadata(metadata=metadata, function_name="skew", aug_img=aug_img, **func_kwargs)
+        return aug_img
```

Comment:
what @zpapakipos tried to say was that these if/elif statements have a lot of duplicate code that we can extract out. see (please make sure the suggestion works on your side):
```suggestion
    if axis == "x-axis":
        data = (1, level, 0, 0, 1, 0)
    elif axis == "y-axis":
        data = (1, 0, 0, level, 1, 0)
    else:
        raise AssertionError(f"Invalid 'axis' value: Got '{axis}', expected 'x-axis' or 'y-axis'")
    
    aug_image = image.transform((w, h),Image.AFFINE, data, resample=Image.BILINEAR)
    imutils.get_metadata(
        metadata=metadata, function_name="skew", aug_img=aug_img, **func_kwargs)
    )
    
    return aug_image
```